sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPump.java

```diff
@@ -61,14 +66,23 @@ void setLastEnqueuedEventProperties(LastEnqueuedEventProperties lastEnqueuedEven
      */
     @Override
     public void close() {
+        // When client receive complete signal will call this method again in clean up method on scheduler thread,
+        // the first dispose will cause a timeout exception. Add a check avoid close twice.
+        if (isDisposed.getAndSet(true)) {
+            return;
+        }
         try {
             client.close();
         } catch (Exception error) {
             LOGGER.atInfo()
                 .addKeyValue(PARTITION_ID_KEY, partitionId)
                 .log("Exception occurred disposing of consumer client.", error);
         } finally {
-            scheduler.dispose();
+            try {
+                ReactorShim.disposeGracefully(scheduler).block(Duration.ofSeconds(1));
```

Comment:
Let's pull out Duration.ofSeconds(1) into a variable and add comment on it saying what it would do and why we are setting it to this value. (Something like grace period for the schedule to be disposed, after this we do force dispose)

Reply:
sure!